client/kubeconfig.go

```diff
@@ -0,0 +1,105 @@
+package client
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/fatih/color"
+)
+
+const DEFAULT_KUBECONFIG_PATH = ".kubeconfig"
+
+type KubeconfigPathOption string
+
+// WithSilent will stop the function from printing information to the console.
+const WithSilent KubeconfigPathOption = "silent"
+
+// LookupKubeconfigPath checks if a kubeconfig file exists, and returns its path.
+//
+// It first checks the "STACC_KUBECONFIG" environment variable, followed
+// by looking in the current directory.
+//
+// Use the WithSilent KubeconfigPathOption to stop this function from
+// logging helpful information directly to the console.
+func LookupKubeconfigPath(options ...KubeconfigPathOption) (string, error) {
+	var kubeconfigPath string
+
+	useSilent := false
+	for _, option := range options {
```

Comment:
This is a lot of code for just a boolean parameter with extra forwards compatibility, maaaybe we should apply YAGNI here and just use a plain boolean parameter? 😅

Reply:
The main reason for using this method was to avoid needing to take a stance on if you want logging on each invocation of the function. The secondary reason was to maintain backwards compatibility, but this isn't relevant as the function name has changed.

I agree that a bool is a saner solution, but don't like the fact that if you see this: `LookupKubeconfigPath(false)`, you have no idea what effect `false` has without reading the documentation/code.

Maybe the best solution is creating two separate functions?
`LookupKubeconfigPath()` and `LookupKubeconfigPathWithLogging()`.
(I've been a bit back and forth between `WithLogging` and `WithReporting`, as "reporting" feels like a more accurate word, but less obvious what it does)

This smells of feature creep, but I want to avoid avoidable breaking changes in the future :sweat_smile:

Reply:
I quite like the two function-approach. Can we just use the current `GetKubeconfigPath()` as the version without logging, and have the "main" function always log? `GetKubeconfigPath()`doesn't do the `.flowrc` check, or the check for whether `.kubeconfig` exists as a directory, but those seem like very niche concerns.

I think I'd prefer `LookupKubeconfigPathWithLogging()` as a name, or `withLogWarnings` or something, "reporting" is more vague, and sounds scary 🫣

Reply:
The main difference between the current `Get[...]()` and `Lookup[...]()` is that `Lookup` returns an error the file doesn't exist. The former is meant to be used to upsert the file, while the latter locates an already existing.

Reply:
Hmm just a thought, but maybe it would be better to have a `GetKubeconfigPath()` and a `ValidateKubeconfigPath(path string)`?

`Get[..]()` would have the same functionailty as now, and `Lookup[...]()` would change name to `Validate[...](path string)` and no longer figure out the path itself. The new functionailty is just a `os.Stat()` with some helpful logging.

Would make the client code a bit more verbose, but the actions are clearer.

Reply:
Alternatively, `Validate()` could stay in this repo, but return the error as a value instead of immediately logging it.

Reply:
Agreed, should be up to the user how they want to validate, considering the varying use-cases.

Will move `Validate()` to the CLI and only leave `Get()` here